Maven Home not resolved with whitespaces in the path

### Describe the bug

If Maven has been unpacked to a location containing a white space (e.g. "C:\Program Files\Maven\apache-maven-3.9.1"), the task cannot resolve it´s Home directory.

### Current behavior

[Issue source code](https://github.com/jfrog/jfrog-azure-devops-extension/blob/875ccd95772f9f4107f90c9054f39a84df81f00c/tasks/JFrogMaven/mavenBuild.ts#L87)
Note that the **.split(' ')[2]** will not return the full path but break on the first whitespace.

**>mvn --version** output:
```
Apache Maven 3.9.1 (2e178502fcdbffc201671fb2537d0cb4b4cc58f8)
Maven home: C:\Program Files\Maven\apache-maven-3.9.1
Java version: 18.0.1.1, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk-18.0.1.1
Default locale: en_US, platform encoding: UTF-8
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
```

**AzureDevOps Pipeline log:**
```
...
Running jfrog-cli from <AgentDir>\_work\_tool\jf\2.36.0\x64\jf.exe
JFrog CLI version: 2.36.0
...
##[debug]M2_HOME=undefined
M2_HOME is not defined. Retrieving Maven home using mvn --version.
The Maven home location: C:\Program
...
Executing JFrog CLI Command:
<AgentDir>\_work\_tool\jf\2.36.0\x64\jf.exe mvn install ...
10:39:40 [Error] couldn't find plexus-classworlds-x.x.x.jar in Maven installation path, please check M2_HOME environment variable
##[debug]task result: Failed
##[error]Error: Command failed: <AgentDir>\_work\_tool\jf\2.36.0\x64\jf.exe mvn install ...
...
```

### Reproduction steps

1. Unpack Maven installation to path that includes a white space
2. Put this path into PATH variable
3. Do NOT define M2_Home, but let the task discover it itself
4. (Re)Start AzureDevOps Agent (to load PATH)
5. Run JFrogMaven task

### Expected behavior

_No response_

### Azure DevOps extension name and version

JFrog 2.5.0

### JFrog CLI version

2.36.0

### Operating system type and version

Windows 10 Enterprise 10.0.19044

### JFrog Artifactory version (if relevant)

_No response_

### JFrog Xray version (if relevant)

_No response_

### JFrog Distribution version (if relevant)

_No response_

--- a/tasks/JFrogMaven/mavenBuild.ts
+++ b/tasks/JFrogMaven/mavenBuild.ts
@@ -84,9 +84,14 @@ function checkAndSetMavenHome(): void {
         let mvnCommand: string = 'mvn --version';
         let res: string = execSync(mvnCommand, { encoding: 'utf-8' } as ExecSyncOptionsWithStringEncoding);
         let mavenHomeLine: string = res.split('\n')[1].trim();
-        let mavenHome: string = mavenHomeLine.split(' ')[2];
-        console.log('The Maven home location: ' + mavenHome);
-        process.env['M2_HOME'] = mavenHome;
+        let regexMatch: RegExpMatchArray | null = mavenHomeLine.match('^Maven\\shome:\\s(.+)');
+        if (regexMatch) {
+            let mavenHomePath: string = regexMatch[1];
+            console.log('The Maven home location: ' + mavenHomePath);
+            process.env['M2_HOME'] = mavenHomePath;
+        } else {
+            console.log('Couldn\'t retrieve Maven home path using "mvn --version" command. Received output: ' + res);
+        }
     }
 }
 

